homeassistant/helpers/template.py

```diff
@@ -141,6 +145,34 @@ def async_render(self, variables: TemplateVarsType = None,
         except jinja2.TemplateError as err:
             raise TemplateError(err)
 
+    def render_with_collect(
```

Comment:
We should not add sync methods unless it's called from a sync context, which I don't think is happening?

Reply:
I think this happens in test cases. Will check later and drop it if not.

Reply:
OK... so this doesn't happen in (some) test cases, although probably it should I think. I have fixed the test cases to use the sync version.

Reply:
I'll close this conversation in favour of the one below...